### PR TITLE
Fix: add cronjob support for annotations, resources, and volumeMounts

### LINKED ISSUE #6421: Cron-task support is incomplete
**Describe the bug**

The cron-task component does not support the volumeMount style of specifiying volumes which has been added to the webservice component.  Hostpath volumes are also therefore not supported.

In addition, annotations and resources traits do not support cron-task.  They seem to be silently accepted but the manifests are not updated correctly, so the trait functionality is not implemented on the cron-task component.

This is due to the fact that they apply to `spec.template` and do not consider `spec.jobTemplate` which exists in `cronjobs.batch`.

**To Reproduce**

Any of the following:

1. Attempt to use `volumeMounts` parameter 
2. Attempt to add a volume of type `hostPath`
3. Attempt to use `annotations` trait with `cron-task` component
4. Attempt to use `resources` trait with `cron-task` component

**Expected behavior**

All of the above scenarios should be supported and correctly apply volumeMounts (including hostPath), annotations and resource constraints.

**Screenshots**

None

**KubeVela Version**

v1.9.7, but applies to all.

**Cluster information**

- Kubernetes Version [e.g. 1.25], but applies to all

**Additional context**
This issue is being added for tracking, I will be creating a pull request shortly with proposed fixes, which will reference this issue.

### PR DESCRIPTION
**Pull request by Kolossi**
### Description of your changes

copilot:all

Adds to `cron-task` support, enabling webservice-style `volumeMounts` (including `hostPath` type), and updates `annotations` and `resources` traits to support `cron-task`.

Fixes #6421 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

I have used the updated definitions on my own cluster.

### Special notes for your reviewer

As discussed on [slack channel](https://cloud-native.slack.com/archives/C01BLQ3HTJA/p1702652404723199).

I could do with som assistance updating the docs.  The PR template mentions kubevela.io, but is it now kubevella.github.io?  I also can't really figure out how the source maps to the published site.  Happy to raise a PR for docs too, but I'm going to need some pointers.

### CODE REVIEW

File: charts/vela-core/templates/defwithtemplate/annotations.yaml [hunk 1]
```diff
@@ -16,17 +16,21 @@ spec:
...
+        	metadata: annotations: annotationsContent
         	if context.output.spec != _|_ && context.output.spec.template != _|_ {
-        		spec: template: metadata: annotations: {
-        			for k, v in parameter {
-        				(k): v
-        			}
-        		}
+        		spec: template: metadata: annotations: annotationsContent
+        	}
+        	if context.output.spec != _|_ && context.output.spec.jobTemplate != _|_ {
```
Comment (FogDong) [inline charts/vela-core/templates/defwithtemplate/annotations.yaml:29]: `&&` short-circuit is not supported in CUE, ref: https://github.com/cue-lang/cue/issues/2232
Please use `if context.output.spec != _|_ if context.output.spec.jobTemplate != _|_` instead. 
Same below.

Comment (Kolossi) [reply charts/vela-core/templates/defwithtemplate/annotations.yaml:29]: thanks @FogDong . Strange I didn't get an error and it worked ok 🤔 

There's also lots of traits and components that use this, not just my change! 😨 

I'll make the change though.

Comment (Kolossi) [reply charts/vela-core/templates/defwithtemplate/annotations.yaml:29]: PR updated

**Review by wonderflow — APPROVED** [review]

**Comment by wonderflow** [comment]
please fix the CI lint

**Comment by Kolossi** [comment]
> please fix the CI lint

I don't understand what to do to fix this.  The "Details" on the failed step seem to show problems in an upstream go module.  It doesn't seem to be reporting issues in the (6) files that are changed in the PR?

If you can let me know what to do, I'll do it :-)

**Review by StevenLeiZhang — APPROVED** [review]
LGTM

**Comment by wonderflow** [comment]
@Kolossi  Maybe try to rebase with master

**Comment by Kolossi** [comment]
Rebased and now the previous fails are passing but it's failing on code coverage and timeout of an e2e multicluster test.

Coverage fail is likely valid, but I'm not knowledgeable enough on kubevela dev to do that. Is there anyone who can assist?

**Comment by FogDong** [comment]
I think it's just because the Github CI machines are not that stable. Let's re-run it.

**Pull request merged**